### DISCUSSION TITLE
[RFC] Provide commonjs locale file in v5

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#38107] [RFCs - archive]
## Summary
v5 产物中新增 `locale` 目录，包含 cjs 版本的 locale 文件。

## Motivation
在 v5 中我们移除了产物中的 `lib` 目录，其中包含了所有 cjs 的文件。因此当前我们提供的格式仅为两种：umd 和 esm。在大部分应用中，这两种格式可以满足需求。但是在同构 SSR 中（如 Next.js)，单独引入 esm 的 locale 文件时会在编译时出错，原因是服务端无法编译 esm。鉴于 locale 文件很多且体积很大，我们不会将其合并到 umd 产物中，所以我们会在产物中单独提供 cjs 版本的 locale 文件。

## Basic Example

```tsx
import React from 'react';
import { ConfigProvider, Modal } from 'antd';
import zhCN from 'antd/locale/zhCN';

export default () => ({
  <ConfigProvider locale={zhCN}>
    <Modal open>test</Modal>
  </ConfigProvider>
})
```

## Detailed Design
在 @ant-design/tools 中添加 locale 的 cjs 编译流程，并移除 es 的 locale 编译。

## How we teach this
在 codemod 中提供迁移脚本，修改用户对 locale 的引入路径。

## Drawbacks
除了 locale 文件需要额外路径引入外，官网的 demo 中也存在大量从 `es` 路径下引入的例子。其中引入类型是不会对编译产生影响的，但是引入其他变量或者函数则会导致无法编译。

## Bonus
使用 father 4 重写 @ant-design/tools。

**Comment by afc163** [comment]
> 但是在同构 SSR 中（如 Next.js)，单独引入 esm 的 locale 文件时会在编译时出错，原因是服务端无法编译 esm

单独引入 esm 的其他文件（非 locale 的文件）不会出问题么？

**Reply by MadCcc** [reply]
下面有补充，引入类型没问题，其他的就是我们不推荐的用法

**Comment by afc163** [comment]
`import zhCN from 'antd/es/locale/zhCN';` 这个还需要保留么

**Reply by MadCcc** [reply]
这个保留的，只新增 cjs 的 locale 文件

**Reply by suxiuxing** [reply]
> 这个保留的，只新增 cjs 的 locale 文件

5.0.0-alpha.31 的 antd/es/locale 不完整
![image](https://user-images.githubusercontent.com/50985188/198339561-cf2443c2-7fbb-4915-ab48-3f23677ab182.png)
